### ISSUE TITLE
SFB tab displays homepage URL as links with incorrect target

### ISSUE THREAD

**Issue by mslw** [psychoinformatics-de/sfb1451-projects-catalog#60]
This occurred locally when adding tabby-datasets, not in the published catalog. Seen for a tabby-dataset with more than one value for its homepage. The SFB tab displays all three links; link text is correct, but the link targets are, respectively, "h", "t", and "t" instead of the full URLs. I suspect an incorrect indexing was introduced in #56 (should be somewhere around references to `https://schema.org/URL`).

**Comment by mslw** [comment]
For a single homepage, catalog behaves correctly (good link target).

**Comment by mslw** [comment]
The inputs generated by https://github.com/sfb1451/tabby-utils/blob/main/load_tabby.py are, in this case (only relevant bits shown, Python syntax):
```
additional_display = {
    'content': {
        'homepage': ['https://example1.com', 'https://example2.com', 'https://example3.com']
    }
}
```

```
additional_display_definitions['SFB1451'] = {
    'values': {
        'homepage': 'https://schema.org/URL',
    }
}
```

The direct cause of the problem in JS code is in this line: https://github.com/psychoinformatics-de/sfb1451-projects-catalog/blob/a8aa5498e4dea95aba8a8748a606639c07f65459/docs/templates/dataset-template.html#L510-L511

This line is executed when content to be displayed is an array. The code expects the definitions to also be an array (`getDefinitionUrl` called with `idx` as third positional argument). However, as we've seen above, the definition is a single string, not an array, the function returns a single letter, and the comparison yields false.

A naive fix of calling `getDefinitionURL` with `null` instead of `idx` solves the issue with URLs (if-statement matches, and the url is rendered as link).

However, that fix repeated across other if-statements completely messes up display of multiple data controllers (in which case, the content is an array of objects)...

<hr>

AFAIK we do not have a formal specification for how the inputs should be formatted, we only have the implementation of JS handler (here) and Python generator (in sfb1451/tabby_utils).

Is the JS code to be fixed (to correctly process an array of things isung a single definition)? Or is the Python input-generation to be improved (by supplying an array of definitions whenever there's an array of values)? I think it's the former.

Do we even have a reasonable case when an array of items (shown in a single table cell) need to use different rendering instructions? It kind-of is how UBERON and NCBITaxon terms are treated currently, but I'm thinking of an alternative solution.

<hr>

Working on this I started thinking about an alternative solution - I'll jot it down elsewhere and link it here if I arrive at something reasonable.

**Comment by mslw** [comment]
This was closed by #62
